apps/marginfi-v2-trading/src/store/uiStore.ts

```diff
@@ -15,31 +17,41 @@ export enum WalletState {
   BUY = "buy",
 }
 
+type TransactionSettings = {
+  broadcastType: TransactionBroadcastType;
+  priorityType: TransactionPriorityType;
+  maxCapType: MaxCapType;
+  maxCap: number;
+};
+
 interface UiState {
   // State
   isWalletAuthDialogOpen: boolean;
   isWalletOpen: boolean;
   lendingMode: LendingModes;
-  priorityFee: number;
   slippageBps: number;
   platformFeeBps: number;
   isActionComplete: boolean;
   previousTxn: PreviousTxn | null;
   isActionBoxInputFocussed: boolean;
   walletState: WalletState;
   isOnrampActive: boolean;
+  broadcastType: TransactionBroadcastType;
+  priorityType: TransactionPriorityType;
+  maxCapType: MaxCapType;
+  maxCap: number;
 
   // Actions
   setIsWalletAuthDialogOpen: (isOpen: boolean) => void;
   setIsWalletOpen: (isOpen: boolean) => void;
   setLendingMode: (lendingMode: LendingModes) => void;
-  setPriorityFee: (priorityFee: number) => void;
   setSlippageBps: (slippageBps: number) => void;
   setIsActionComplete: (isActionSuccess: boolean) => void;
   setPreviousTxn: (previousTxn: PreviousTxn) => void;
   setWalletState: (walletState: WalletState) => void;
   setIsActionBoxInputFocussed: (isFocussed: boolean) => void;
   setIsOnrampActive: (isOnrampActive: boolean) => void;
+  setTransactionSettings: (settings: TransactionSettings) => void;
```

Comment:
Why have TransactionSettings setter in 1 but the actual type separated in multiple vars?

Reply:
good you bring this up, I'll keep it in the back of my head. Depending on the sdk refactor I might combine this into one ✅